src/overlays/actors/ovl_En_Bat/z_en_bat.c

```diff
@@ -33,99 +40,522 @@ const ActorInit En_Bat_InitVars = {
+    /* Deku bubble    */ DMG_ENTRY(1, BAD_BAT_DMGEFF_NONE),
+    /* Deku launch    */ DMG_ENTRY(2, BAD_BAT_DMGEFF_NONE),
+    /* UNK_DMG_0x12   */ DMG_ENTRY(0, BAD_BAT_DMGEFF_STUN),
+    /* Zora barrier   */ DMG_ENTRY(0, BAD_BAT_DMGEFF_ELECTRIC),
+    /* Normal shield  */ DMG_ENTRY(0, BAD_BAT_DMGEFF_NONE),
+    /* Light ray      */ DMG_ENTRY(0, BAD_BAT_DMGEFF_NONE),
+    /* Thrown object  */ DMG_ENTRY(1, BAD_BAT_DMGEFF_NONE),
+    /* Zora punch     */ DMG_ENTRY(1, BAD_BAT_DMGEFF_NONE),
+    /* Spin attack    */ DMG_ENTRY(1, BAD_BAT_DMGEFF_NONE),
+    /* Sword beam     */ DMG_ENTRY(0, BAD_BAT_DMGEFF_NONE),
+    /* Normal Roll    */ DMG_ENTRY(0, BAD_BAT_DMGEFF_NONE),
+    /* UNK_DMG_0x1B   */ DMG_ENTRY(0, BAD_BAT_DMGEFF_NONE),
+    /* UNK_DMG_0x1C   */ DMG_ENTRY(0, BAD_BAT_DMGEFF_NONE),
+    /* Unblockable    */ DMG_ENTRY(0, BAD_BAT_DMGEFF_NONE),
+    /* UNK_DMG_0x1E   */ DMG_ENTRY(0, BAD_BAT_DMGEFF_NONE),
+    /* Powder Keg     */ DMG_ENTRY(1, BAD_BAT_DMGEFF_NONE),
 };
 
-// sColChkInfoInit
-static CollisionCheckInfoInit D_80A44A4C = { 1, 15, 30, 10 };
+static CollisionCheckInfoInit sColChkInfoInit = { 1, 15, 30, 10 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80A44A54[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_S8(hintId, 96, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneForward, 3000, ICHAIN_CONTINUE),
     ICHAIN_F32_DIV1000(gravity, -500, ICHAIN_CONTINUE),
     ICHAIN_F32(targetArrowOffset, 2000, ICHAIN_STOP),
 };
 
-#endif
+static Gfx* sWingsDLs[] = {
+    gBadBatWingsFrame0DL, gBadBatWingsFrame1DL, gBadBatWingsFrame2DL, gBadBatWingsFrame3DL, gBadBatWingsFrame4DL,
+    gBadBatWingsFrame5DL, gBadBatWingsFrame6DL, gBadBatWingsFrame7DL, gBadBatWingsFrame8DL,
+};
+
+#define BAD_BAT_MAX_NUMBER_ATTACKING 3
+
+s32 sNumberAttacking; //!< Limit number attacking player to at most `BAD_BAT_MAX_NUMBER_ATTACKING`
+s32 sAlreadySpawned;  //!< used for those spawned with room -1 in Graveyard to avoid respawn on room change
+
+void EnBat_Init(Actor* thisx, PlayState* play) {
+    EnBat* this = THIS;
+
+    Actor_ProcessInitChain(thisx, sInitChain);
+    Collider_InitAndSetSphere(play, &this->collider, thisx, &sSphereInit);
+    this->collider.dim.worldSphere.radius = sSphereInit.dim.modelSphere.radius;
+    CollisionCheck_SetInfo(&thisx->colChkInfo, &sDamageTable, &sColChkInfoInit);
+    ActorShape_Init(&thisx->shape, 2000.0f, ActorShadow_DrawCircle, 25.0f);
+
+    this->animationFrame = Rand_ZeroOne() * 9.0f;
+
+    this->paramFlags = BAD_BAT_GET_PARAMFLAGS(thisx);
+    this->switchFlag = BAD_BAT_GET_SWITCHFLAG(thisx);
+    thisx->params = BAD_BAT_GET_TYPE(thisx);
+
+    thisx->depthInWater = BGCHECK_Y_MIN;
+    Actor_SetFocus(thisx, 20.0f);
+
+    if (sAlreadySpawned) {
+        Actor_MarkForDeath(thisx);
+    } else if (EnBat_IsGraveyardOnSecondDay(play)) {
+        if (Flags_GetSwitch(play, this->switchFlag)) {
+            Actor_MarkForDeath(thisx);
+        } else {
+            this->actor.room = -1;
+        }
```

Comment:
Ahhhh its hard coded damn it, here I was thinking it would be a param deciding if it set the switch